presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/SingleMarkDistinctToGroupBy.java

```diff
@@ -82,6 +82,9 @@ public Pattern getPattern()
             return Optional.empty();
         }
 
+        if (!child.getOrderBySymbols().isEmpty()) {
```

Comment:
They will be the same, but the logic here is that if the node has order by don't optimize MarkDistinct, so it feels more straight forward to me to use MarkDistinctNode.getOrderBySymbols. I don't have strong opinions. If you really like parent more, I can change.